src/Compilers/CSharp/Portable/Binder/Binder_Initializers.cs

```diff
@@ -152,42 +152,34 @@ internal struct ProcessedFieldInitializers
                     var syntaxRef = initializer.Syntax;
                     Debug.Assert(syntaxRef.SyntaxTree.Options.Kind != SourceCodeKind.Regular);
 
-                    var initializerNode = (CSharpSyntaxNode)syntaxRef.GetSyntax();
-
-                    if (binderFactory == null)
-                    {
-                        binderFactory = compilation.GetBinderFactory(syntaxRef.SyntaxTree);
-                    }
-
-                    Binder scriptClassBinder = binderFactory.GetBinder(initializerNode);
-                    Debug.Assert(((ImplicitNamedTypeSymbol)scriptClassBinder.ContainingMemberOrLambda).IsScriptClass);
-
-                    if (firstDebugImports == null)
+                    if (parentBinder == null)
                     {
-                        firstDebugImports = scriptClassBinder.ImportChain;
+                        var syntaxTree = syntaxRef.SyntaxTree;
+                        var binderFactory = compilation.GetBinderFactory(syntaxTree);
+                        var compilationUnit = (CompilationUnitSyntax)syntaxTree.GetRoot();
+                        var executableCodeBinder = new ExecutableCodeBinder(compilationUnit, scriptInitializer, binderFactory.GetBinder(compilationUnit));
+                        parentBinder = executableCodeBinder.GetBinder(compilationUnit);
+                        firstDebugImports = parentBinder.ImportChain;
                     }
 
-                    Binder parentBinder = new ExecutableCodeBinder((CSharpSyntaxNode)syntaxRef.SyntaxTree.GetRoot(), scriptInitializer, scriptClassBinder);
-
+                    var syntax = (CSharpSyntaxNode)syntaxRef.GetSyntax();
                     BoundInitializer boundInitializer;
                     if ((object)fieldSymbol != null)
                     {
                         boundInitializer = BindFieldInitializer(
-                            new LocalScopeBinder(parentBinder).WithAdditionalFlagsAndContainingMemberOrLambda(parentBinder.Flags | BinderFlags.FieldInitializer, fieldSymbol),
+                            parentBinder.WithAdditionalFlagsAndContainingMemberOrLambda(BinderFlags.FieldInitializer, fieldSymbol),
```

Comment:
Why remove the ```LocalScopeBinder```?  I seem to recall that we used to have problems with code walking up the chain looking for one and not finding any.  This might have been inserted to prevent that.